progs/list-names/main.go

```diff
@@ -18,5 +19,17 @@ func main() {
 	}
 	defer f.Close()
 
-	fmt.Println("I should be processing the input file now!")
+	//####################################
+
+	fileScanner := bufio.NewScanner(f)
+
+	output := ""
+	for fileScanner.Scan() {
+		line := fileScanner.Text()
+		if line[0:1] == "@" {
+			output = output + line[1:] + "\n"
```

Comment:
This concatenation takes time proportional to the input strings. It turns this scan into a quadratic time algorithm rather than a linear one.